controllers/limitador_cluster_envoyfilter_controller.go

```diff
@@ -46,6 +46,7 @@ type LimitadorClusterEnvoyFilterReconciler struct {
 
 //+kubebuilder:rbac:groups=networking.istio.io,resources=envoyfilters,verbs=get;list;watch;create;update;patch;delete
 //+kubebuilder:rbac:groups=gateway.networking.k8s.io,resources=gateways,verbs=get
+//+kubebuilder:rbac:groups=gateway.networking.k8s.io,resources=gateways/finalizers,verbs=update
```

Comment:
I don't get it. Is that because we create the EnvoyFilter CR? Our controller does not touch the gateway finalizer ever. Is there any path (perhaps through Istio?) that requires this permission? Effectively what it does is grating permission to _our_ `controller-manager` Service Account to update this kind of resource.

Reply:
Hmmm. Nothing to do with Istio.

It looks like the issue is at this line: https://github.com/Kuadrant/kuadrant-operator/blob/982c978619675d912dd55b32fdda09ab823609ae/controllers/limitador_cluster_envoyfilter_controller.go#L155

And more concretely: https://github.com/Kuadrant/kuadrant-operator/blob/982c978619675d912dd55b32fdda09ab823609ae/pkg/reconcilers/base_reconciler.go#L189

Reply:
Yes, sorry, I should have pointed it out explicitly... That's indeed what I think is the issue in [OpenShift](https://kubernetes.io/docs/reference/access-authn-authz/admission-controllers/#ownerreferencespermissionenforcement)